Tools/automation.ps1

```diff
@@ -0,0 +1,791 @@
+    {
+        ## Verifies connection to Azure was successful.
+        If($Azerror)
+        { 
+            ## Connection to Azure failed.
+            Write-Host "  Failed to connect to Azure Environment...`n  $Azerror" -ForegroundColor Red
+            Return $false
+        }
+
+        ## Retrieves the connected Azure Subscription
+        $AzSubscriptionName = $(Get-AzContext).Subscription.Name
+        
+        ## Connection to Azure was successful.
+        Write-Host "  Connected to Azure Environment: `n    Subscription ID: $AzSubscriptionName"
+        Return $true
+    }
+}
+
+Function New-AzureResourceGroup
+{
+    Param(
+        [Parameter(Position=1, Mandatory=$true)] [string]$AzResourceGroupName
+    )
+    Begin
+    {
+        ## Creating a line break from previous steps
+        Write-Host "`n"
+    }
+    Process
+    {
+        ## Determines if the Resource Group already exists
+        $Result = Get-AzResourceGroup -Name $AzResourceGroupName -ErrorAction SilentlyContinue -ErrorVariable RGerror -InformationAction SilentlyContinue -WarningAction SilentlyContinue
+
+        IF($RGerror)
+        { 
+            ## Resource Group does not already exist, creating a new Resource Group
+            Write-Host "Creating Resource Group ($AzResourceGroupName) in the $AzLocation region..."
+            $Result = New-AzResourceGroup -Name $AzResourceGroupName -Location $AzLocation 
+        }
+    }
+    End
+    {
+        IF(!$RGerror)
+        {
+            ## Resource Group already exists, do nothing
+            Write-Host "Resource Group $AzResourceGroupName already exists, will not recreate..." -ForegroundColor Yellow
+            Return $true
+        }
+        ElseIF(Get-AzResourceGroup -Name $AzResourceGroupName -ErrorAction SilentlyContinue -WarningAction SilentlyContinue -InformationAction SilentlyContinue)
+        {
+            ## Resource Group was created Successfully 
+            Write-Host "  Resource Group ($AzResourceGroupName) in the $AzLocation region was created successfully"
+            Return $true
+        }
+        else 
+        {
+            ## Resource Group failed to be created 
+            Write-Host "  Resource Group ($AzResourceGroupName) in the $AzLocation region failed to be created" -ForegroundColor Red 
+            Return $false
+        }
+    }
+}
+
+Function New-ARMParameterObject
+{
+    param(
+        [Parameter(Position=0, Mandatory=$true)] [string]$ParameterFolderPath,
+        [Parameter(Position=1, Mandatory=$true)] [string]$TemplateFolderPath,
+        [Parameter(Position=2, Mandatory=$false)][string]$Index,
+        [Parameter(Position=3, Mandatory=$true)] [string]$ResourcePrefix,
+        [Parameter(Position=4, Mandatory=$true)] [string]$AzLocation
+    )
+    Begin
+    {
+        ## The Names that are to be assigned to each resource.
+        $AppInsightsName    = $($ResourcePrefix + "appinsight" + $Index)            # Name that will be assigned to the Azure AppInsights.
```

Comment:
place comments on their own line.

Reply:
Removed comments at the end of cmds. Moved the commented code to it's own line.